trader/trader.go

```diff
@@ -615,30 +616,29 @@ func (t *Trader) setExistingOffers(sellingAOffers []hProtocol.Offer, buyingAOffe
 	t.sellingAOffers, t.buyingAOffers = sellingAOffers, buyingAOffers
 }
 
-func (t *Trader) countOfferChangeTypes(offers []txnbuild.Operation) (numCreate int, numUpdate int, numDelete int, e error) {
-	var offerXdr xdr.Operation
+func countOfferChangeTypes(offers []*txnbuild.ManageSellOffer) (int, int, int, error) {
+	numCreate, numDelete, numUpdate := 0, 0, 0
 	for _, o := range offers {
-		offerXdr, e = o.BuildXDR()
-		if e != nil {
-			return 0, 0, 0, fmt.Errorf("could not convert op to offer: %s", e)
+		if o == nil {
+			return 0, 0, 0, fmt.Errorf("offers were not of expected type ManageSellOffer")
 		}
 
-		manageSellOp := offerXdr.Body.ManageSellOfferOp
-		if manageSellOp == nil {
-			return 0, 0, 0, fmt.Errorf("offers were not of expected type ManageSellOffer")
+		opAmount, e := strconv.ParseFloat(o.Amount, 64)
+		if e != nil {
+			return 0, 0, 0, fmt.Errorf(" invalid operation amount (%s) could not be parsed as float", o.Amount)
```

Comment:
`..for operation at index %d: %v", ..., i, o` so we print the whole offer as well for easier debugging